HIP/hip-655.md

```diff
@@ -0,0 +1,106 @@
+updated: 2023-01-20, 2023-01-24
+---
+
+## Abstract
+
+[HIP-23](https://hips.hedera.com/hip/hip-23) allowed an account owner to open up a specified number of automatic token-association slots into which others can transfer tokens. However, this capability doesn't allow for a pure "airdrop" of tokens. This HIP provides an additional mechanism for senders to airdrop tokens to other accounts, while allowing individual account holders to maintain control over what shows up in their account balances.
+
+## Motivation
+
+The current implementation of the Hedera Token Service does not allow airdrops of arbitrary tokens to any account. [HIP-23](https://hips.hedera.com/hip/hip-23) lists several disadvantages of arbitrary airdrops - such as exposing an innocent account holder to reputational or legal or spam risk of being associated with unwanted/unsolicited tokens, as well as DDoS risks for the network. And it proposes opt-in token associations. Further, [HIP-367](https://hips.hedera.com/hip/hip-367) removes the limit on token associations. Even with these enhancements, users still cannot airdrop tokens using the Hedera network today.
+
+## Rationale
+
+Token airdrops are very common in the blockchain industry and many legitimate use cases require airdropping of tokens. This HIP attempts to provide a mechanism to enable these airdrops while still protecting the account owners, and imposing dis-incentivizing costs to the senders of the airdropped tokens.
+
+## User Stories
+
+### Terminology
+
+The following terminology is used in the rest of this Specification
+
+- Airdrop-Sender: the owner of a token that they wish to send to somebody without the explicit association/permission from the intended recipient
+- Airdrop-Recipient: the intended recipient of the above airdrop
+- Node-Operator: the operator of a node in the Hedera network
+    
+### User stories
+    
+- As an Airdrop-Sender, I want to send arbitrary tokens (Fungible and/or Non-fungible) to Airdrop-Recipients so that I can create interesting business use cases
+    1. I understand that the recipients of the airdrop may not wish to accept these tokens and I am willing to pay for the network resources used while they make their decision.
+    2. If the intended Airdrop-Recipients do not accept my transfer, I want to be able to restore those tokens to my account so that I can do whatever I wish with those tokens.
+- As an Airdrop-Recipient, I want to receive arbitrary tokens (Fungible and/or Non-fungible) from the Airdrop-Senders so that I can accumulate those in my account.
+    1. I want to achieve the above while having explicit control over what tokens actually show up in my account.
+    2. I want to achieve the above without being subject to the reputational/legal/spam risk of being associated with unwanted tokens
+    3. I want to achieve the above without having to be forced to pay for the storage/renewal of these unwanted tokens unless I explicitly accept those.
```

Comment:
> 4. I want to reject an airdrop and remove it from my inbox without paying any fees to the network.

5. I want to automatically approve all tokens sent to me with a treasury account id / account ids I define
6. I want to automatically reject all tokens sent to me that aren't associated

Reply:
7. I want to automatically reject all tokens sent to me with a treasury account id / account ids I define

Reply:
I wonder, could we leave these as a future enhancement or if we need to flesh these out now? Seems like someone could also offer this as a service.